src/compiler/commandLineParser.ts

```diff
@@ -374,10 +374,10 @@ namespace ts {
       * Read tsconfig.json file
       * @param fileName The path to the config file
       */
-    export function readConfigFile(fileName: string): { config?: any; error?: Diagnostic }  {
+    export function readConfigFile(fileName: string, system: System = sys): { config?: any; error?: Diagnostic }  {

```

Comment:
I agree completely (and have followed up by doing so). But we should acknowledge this is a breaking change to a public API.